### ISSUE TITLE
private[this] var in class constructor disallows mutation

### ISSUE THREAD

**Issue by scabug** [scala/bug#6165]
```scala
scala> class PT(private[this] var f:String) { f = f + "hellO"; def ff = f }
defined class PT

scala> (new PT("")).ff
res0: String = ""

scala> class P(private var f:String) { f = f + "hellO"; def ff = f }
defined class P

scala> (new P("")).ff
res1: String = hellO
```

Is this intentional?

**Comment by SethTisue** [comment]
still present in 2.12.4

**Comment by som-snytt** [comment]
Fixed in 2.13, but I didn't see where.

**Comment by som-snytt** [comment]
I didn't see that it duplicates scala/bug#6880
